### ISSUE TITLE
Error when trying to restage data that has not yet been tiered

### ISSUE THREAD

**Issue by JustinKyleJames** [irods/irods_capability_storage_tiering#162]
I am getting an error like the following when storage tiering kicks off.  According to Jason this is because the data object has not tiered yet and therefore doesn't have metadata attached to it.
```
Feb 23 14:31:41 pid:15851 NOTICE: Failed to restage data object [/zone/home/user1/iget_tiny] for resource [col01_ufs_resc] Exception: [iRODS Exception:
    file: /irods_plugin/storage_tiering.cpp
    function: std::string irods::storage_tiering::get_group_name_by_replica_number(rcComm_t *, const std::string &, const std::string &, const std::string &)
    line: 773
    code: -808000 (CAT_NO_ROWS_FOUND)
    message:
        failed to fetch group name by resource and replica number
```

Here is the relevant metadata on the UFS resource:

```
$ imeta ls -R col01_ufs_resc
AVUs defined for resource col01_ufs_resc:
attribute: irods::storage_tiering::group
value: col01_tier_group
units: 0
----
attribute: irods::storage_tiering::object_limit
value: 50
units: 
----
attribute: irods::storage_tiering::query
value: col01_archive_query
units: specific
----
attribute: irods::storage_tiering::time
value: 3600
units: 

```
Here is the metadata on the tier 1 resource:

```
$ imeta ls -R s3_resc
AVUs defined for resource s3_resc:
attribute: irods::storage_tiering::group
value: col01_tier_group
units: 1
----
attribute: irods::storage_tiering::group
value: por01_tier_group
units: 1
----
attribute: irods::storage_tiering::preserve_replicas
value: true
units: 
```

Here is the metadata on the data object:

```
$ imeta ls -d /zone/home/user1
AVUs defined for dataObj /zone/home/user1:
attribute: irods::access_time
value: 1614081551
units:

**Comment by alanking** [comment]
I cannot reproduce this with iRODS 4.3.3 and plugin version 4.3.3.0.

It looks like there are a few things going on in this report and I feel that I do not actually understand what the issue is. Here is the setup that I'm seeing...

There is a data object with a replica on the lowest tier of a storage tiering group. Before it has been migrated to a higher tier (i.e. tiered out) for the first time, read the data so that it will be (potentially) scheduled for restaging. We expect to see an error in the log regarding a failed restage. We can also see that the S3 resource shown above serves as tier 1 for 2 tiering groups, which I did not include in my setup. I figured this was irrelevant given that this issue has to do with restaging.

I did that and did not see any errors in the log. The replica stayed put where it was, and I was able to read the data with no problems. I wonder whether this issue was incidentally resolved as a result of work done for restaging in the last few years.

Please let me know if I'm misunderstanding something. I'm not sure that we want to add a test for this since it would require reading log messages unless there's some other failure indicator.

**Comment by alanking** [comment]
Thinking about this more, I could see an issue in the same class of problem as #72 which involves data landing in a middle tier (e.g. tier 1) and then attempting to restage it to a lower tier. This would likely exhibit the problem described here for the reasons described here.

I think that the restaging would fail because it is not known to which tier group the data belongs. If we had to assign it to a group, we would have to put it in all the tier groups to which the host resource belongs. This problem - or something like it - would exist for initial tier resources which belong to multiple groups as well.

**Comment by korydraughn** [comment]
> I wonder whether this issue was incidentally resolved as a result of work done for restaging in the last few years.

That's a real possibility.

If you cannot reproduce it or feel there's not enough information to fully understand the problem, then let's close it.

If this is still a problem, someone will report it.

**Comment by alanking** [comment]
The log message reported in both of these issues was changed along with the surrounding behavior in commit https://github.com/irods/irods_capability_storage_tiering/commit/965281a839b212679f3de89c506d527e3b6976e7, which resolved #232. This is fixed as of 4.3.2.0. Closing.
